### DISCUSSION TITLE
define constants for DataObject classes

### DISCUSSION THREAD

**Discussion by langert** [pimcore/pimcore#16428] [Q&A / Support]
Hi,
is there a "pimcore way" to define constants for a DataObject class, without to overwrite the whole class?

**Comment by beheist** [comment]
@langert You could define a trait and put them in there, and then add the trait to the object class via the class editor.
Also depends a bit on where/how you want to use them, I'd say.

**Reply by langert** [reply]
in which php version you can define constants in traits?

**Reply by beheist** [reply]
Huh, you are right, doesn't seem to be possible. Never tried it myself, so it was just an idea ;)

**Reply by beheist** [reply]
No wait, it seems to be possible in 8.2: https://php.watch/versions/8.2/constants-in-traits
